### PR TITLE
Remove storedFilters from the dependency array in PersistentFilter's useEffect

### PR DESCRIPTION
**Pull request by myarmolinsky**
Remove storedFilters from the dependency array in PersistentFilter's useEffect

Change-type: patch
Signed-off-by: Matthew Yarmolinsky <matthew-timothy@balena.io>

---

##### Contributor checklist

- [ ] I have regenerated jest snapshots for any affected components with `npm run generate-snapshots`

##### Reviewer Guidelines

- When submitting a review, please pick:
  - '_Approve_' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '_Request Changes_' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '_Comment_' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)

---

### CODE REVIEW

File: src/extra/AutoUI/index.tsx [hunk 1]
```diff
@@ -303,11 +303,13 @@ export const AutoUI = <T extends AutoUIBaseResource<T>>({
 												views={views}
 												autouiContext={autouiContext}
 												changeFilters={(updatedFilters) => {
-													setFilters(updatedFilters);
-													if (onChange) {
-														onChange({
-															filters: updatedFilters,
-														});
+													if (!isEqual(updatedFilters, filters)) {
```
Comment (thgreasi) [inline src/extra/AutoUI/index.tsx:306]: How about moving this equality check in the `onFiltersUpdate?.(storedFilters);` useEffect?
If it works then imo it's a more appropriate place to reduce unnecessary changes.

Comment (myarmolinsky) [reply src/extra/AutoUI/index.tsx:306]: Seems to work

**Comment by myarmolinsky** [comment]
Closing in favor of https://github.com/balena-io/balena-ui/pull/5442
